components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -398,6 +398,18 @@ private void setAdaptiveAuthExecutionSupervisor() {
             }
         }
 
+        String timeOutEnabledString = IdentityUtil.getProperty(
+                FrameworkConstants.AdaptiveAuthentication.CONF_EXECUTION_SUPERVISOR_TIMEOUT_ENABLE);
+        boolean timeOutEnabled = FrameworkConstants.AdaptiveAuthentication
+                .DEFAULT_EXECUTION_SUPERVISOR_TIMEOUT_ENABLE;
+        if (StringUtils.isNotBlank(timeOutEnabledString)) {
+            try {
+                timeOutEnabled = Boolean.parseBoolean(timeOutEnabledString);
```

Comment:
Boolean.parseBoolean will return false for anything other than `true`. Is it expected here?

Reply:
Yes, we have set the default value as `true` in the default.json. If the user has configured it in invalid way, then this will return false which is expected